.github/workflows/java-build-release.yaml

```diff
@@ -0,0 +1,83 @@
+name: Release Artifact Publisher
+
+on:
+  workflow_call:
+    inputs:
+      release_version:
+        required: true
+        type: string
+      repository_url:
+        required: true
+        type: string
+      gradle_command:
+        required: true
+        type: string
+      branch:
+        required: true
+        type: string
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    env:
+      GH_USER_NAME: ${{ github.actor }}
+
+    steps:
+      - name: Set ref
+        run: |
+            echo "REF=$([[$GITHUB_EVENT_NAME == 'release']] && echo refs/tags/${{ github.event.release.tag_name }} || echo ${{ inputs.branch }})" >> $GITHUB_ENV
+
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+         ref: ${{ env.REF }}
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '21'
```

Comment:
I recommend using input variables for the Java version. Different services can use other versions.